### DISCUSSION TITLE
How works node-linker=hoisted in PNPM workspace?

### DISCUSSION THREAD

**Discussion by sleaper** [pnpm/pnpm#5779] [Q&A] [answered]
Hi guys,
I want to ask you. I have monorepo with this folder structure:
```
root
├── backend
├── mobile-app
├── web-extension
└── shared
```

In each subfolder, there is package.json with packages that I need. In the root, I have `.npmrc`, in which I have 
`node-linker=hoisted`. I thought that it means that all packages from all subfolders (backend, mobile-app...) would be hoisted to the root. But actually, just some of the packages are and some of them are not. How does this work? E.g when I run `pnpm add jsc-android` in mobile-app it actually downloads it in root node_modules. On the other hand, when I run`pnpm add react-native-mmkv` in mobile-app, It downloads to node_module in mobile-app. How does pnpm work which node_module to choose?
I need some packages to not be hoisted.
If you have any questions feel free to ask!

**Comment by zkochan** [comment, marked as answer]
Everything is hoisted to the root node_modules of the workspace (which is the directory where you have the `pnpm-workspace.yaml` file). Unless different versions of the same dependency are present in multiple projects in the workspace. In that case one of the versions will be hoisted to the root of the workspace. The other ones will be hoisted just to the root of the project.

**Reply by bvincent** [reply]
Hello,

With `node-linker=hoisted`, is it possible to set nohoist option for some dependencies ? Negative `hoist-pattern` configuration seems not work with `node-linker=hoisted`.

**Reply by djbalin** [reply]
@bvincent did you find a way to do this? struggling with the same: how to exclude some packages from being hoisted with `node-linker=hoisted` set
